cirq/optimizers/merge_single_qubit_gates_test.py

```diff
@@ -141,3 +149,26 @@ class UnsupportedDummy(cirq.Gate):
     cirq.MergeSingleQubitGates().optimize_circuit(circuit)
 
     assert circuit == c_orig
+
+
+def test_rewrite():
+    q0 = cirq.LineQubit(0)
+    q1 = cirq.LineQubit(1)
+    circuit = cirq.Circuit.from_ops(
+        cirq.X(q0),
+        cirq.X(q1),
+        cirq.Y(q0),
+        cirq.CZ(q0, q1),
+        cirq.Y(q1),
+    )
+    cirq.MergeSingleQubitGates(
+        rewriter=lambda ops: cirq.H(ops[0].qubits[0])
+    ).optimize_circuit(circuit)
+    cirq.DropEmptyMoments().optimize_circuit(circuit)
+
+    cirq.testing.assert_same_circuits(circuit, cirq.Circuit.from_ops(
+        cirq.H(q0),
+        cirq.H(q1),
+        cirq.CZ(q0, q1),
+        cirq.H(q1),
+    ))
```

Comment:
Maybe add tests for `merge_single_qubit_gates_into_phased_x_z()`, too.

For example, we could verify that it optimizes Y down to a single gate and identity to no gates at all, but H to two(*) gates.

--
(*) The invariant which makes it clear is the equality of the diagonal elements. It's true for all PhasedXGates and false for the Hadamard.

Reply:
Done.